pkg/codegen/pcl/binder_schema.go

```diff
@@ -301,6 +286,23 @@ func (b *binder) schemaTypeToType(src schema.Type) model.Type {
 			return model.NewUnionTypeAnnotated(types, src)
 		}
 		return model.NewUnionType(types...)
+	case *schema.ResourceType:
+		if t, ok := b.schemaTypes[src]; ok {
+			return t
+		}
+
+		properties := map[string]model.Type{}
+		objType := model.NewObjectType(properties, src)
+		b.schemaTypes[src] = objType
+		for _, prop := range src.Resource.Properties {
+			typ := prop.Type
+			if !prop.IsRequired() {
+				typ = &schema.OptionalType{ElementType: typ}
+			}
+
```

Comment:
Do we need to do any input conversions (`model.InputType`)? I don't understand where the `schema.Propertie.Plain` flag is encoded.

Reply:
I cannot honestly say I know enough about our binding and types to be sure, please let me know if this helps you answer:

This code path and the types we generate here should only appear in output positions: scope traversals/accessors off of a resource node.

The properties we enumerate over / the `typ` here should encode the `plain` flag as well as `default`, `const` fields.